OpenUpSA/wazimap-ng-ui
# hiding category and subcategory headings when no indicators available

## Description
hiding empty categories & sub-categories

## Related Issue
https://wazimap.atlassian.net/browse/WNCM-263

## How is it tested automatically?
* added test steps into `__tests__/gui/rich_data.feature`

## How should a reviewer test it locally
* find a category or a subcategory that does not have any indicators.
* hint : when you go down to lowest level e.g. ward/main place you can often find categories or subcategories without any data
* confirm that categories & subcategories without indicators are not displayed 
* confirm that the categories without indicators are not listed in the left hand-side nav menu

## Screenshots
before : 
![image](https://user-images.githubusercontent.com/53019884/182375897-74406232-843d-494d-91d4-53b50925c70f.png)

after : 
![image](https://user-images.githubusercontent.com/53019884/182375996-37aa7efe-091f-4efa-8db8-fea6f6324a17.png)


## Changelog

### Added

### Updated
* `src/js/profile/profile_loader.js`, `src/js/profile/category.js`, `src/js/profile/subcategory.js` to hide empty categories & subcategories

### Removed


## Checklist


### Pull Request


### Code Quality


### Testing
